lunatrace/bsl/backend/src/models/vulnerability-dependency-tree/index.ts

```diff
@@ -182,19 +189,22 @@ export default class VulnerabilityDependencyTree {
 
       const triviallyUpdatable = isReleaseTriviallyUpdateable(packageReleaseVulnerabilities);
 
-      const existingRelease = vulnerableReleasesById.get(node.release_id);
-      packageReleaseVulnerabilities.forEach((releaseVulnerability) => {
-        const updatedRelease = createOrMergeVulnerablePackageRelease(
+      // loop each vulnerability on the node
+      packageReleaseVulnerabilities.forEach((releaseVulnerability, index) => {
+        const existingRelease = vulnerableReleasesById.get(node.release_id);
+        const isFirstVulnOnNode = index === 0;
+
+        const upsertedRelease = createOrMergeVulnerablePackageRelease(
           existingRelease,
           node.release,
           releaseVulnerability,
-          vulnerableNodeChains,
+          isFirstVulnOnNode ? chains : [],
```

Comment:
this and a one line change in this function that amalgamates the chains are the only important changes in this pr

Reply:
I don't think we need `isFirstVulnOnNode` anymore now that `existingRelease` is properly set.

Reply:
yeah we still do, we can only do this once per node and the same node may have multiple vulnerabilities. Doing it again for each vulnerability would produce duplicate chains on the release